### PR TITLE
test: Code Coverage: Notifications Page #1 

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 0667180</samp>

Added mock data and unit tests for the notification settings feature. Implemented a method to select or deselect all notification events in `NotificationsPage`. Removed a debugging statement from `notifications.page.ts`.

### <samp>🤖 Generated by Copilot at 0667180</samp>

> _You can't escape the notifications of doom_
> _They haunt you and your delegatee too_
> _You mock them with your data and your tests_
> _But they will always find a way to get through_

### <samp>🤖 Generated by Copilot at 0667180</samp>

*  Add `toggleEvents` method to `NotificationsPage` component to allow user to select or deselect all email events ([link](https://github.com/fylein/fyle-mobile-app/pull/2448/files?diff=unified&w=0#diff-32a367e9037385cdfd6b445a0a107423456bdcad37833cc3c36932aaf9ef4529R294-R298))
*  Import `EmailEvents` enum from `org-settings.model` to use it in mock data for `orgSettingsWithUnsubscribeEvent` ([link](https://github.com/fylein/fyle-mobile-app/pull/2448/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951L2-R2))
*  Add mock data for `orgSettingsWithUnsubscribeEvent` to test the scenario where the admin has unsubscribed from some email events ([link](https://github.com/fylein/fyle-mobile-app/pull/2448/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951R1349-R1356))
*  Add mock data for `notificationDelegateeSettings1`, `notificationDelegateeSettings2`, and `notificationDelegateeSettings3` to test different combinations of notification settings for the user and the delegatee ([link](https://github.com/fylein/fyle-mobile-app/pull/2448/files?diff=unified&w=0#diff-de467ea1b214e32a89452896f193a5769a02286b73795b8723305087a9ad59c9R686-R712))
*  Add unit tests for `NotificationsPage` component in `notifications.page.spec.ts` file, using the mock data and providers ([link](https://github.com/fylein/fyle-mobile-app/pull/2448/files?diff=unified&w=0#diff-9a90ed5661f0bc4fea1b33e636b77a68a4ba9e9c1959d9608079893753be9985L1-R209))

## Clickup
https://app.clickup.com/t/85ztznwfy

## Code Coverage
`32.25% Statements 40/124`
`36.66% Branches 11/30`
`28.2% Functions 11/39`
`32.75% Lines 38/116`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/notifications/notifications.page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,210 @@
...
+import { of } from 'rxjs';
+import { apiEouRes } from 'src/app/core/mock-data/extended-org-user.data';
+import { orgSettingsData } from 'src/app/core/test-data/accounts.service.spec.data';
+import { notificationEventsData } from 'src/app/core/mock-data/notification-events.data';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { orgSettingsWithUnsubscribeEvent } from 'src/app/core/mock-data/org-settings.data';
+import { emailEvents } from 'src/app/core/mock-data/email-events.data';
 
-xdescribe('NotificationsPage', () => {
+export class NavMock {
```
Comment (OmkarJ13) [inline src/app/fyle/notifications/notifications.page.spec.ts:25]: Why exporting this?

File: src/app/fyle/notifications/notifications.page.spec.ts [hunk 2]
```diff
@@ -1,26 +1,210 @@
...
+          provide: OrgSettingsService,
+          useValue: orgSettingsServiceSpy,
+        },
+        {
+          provide: Router,
+          useValue: routerSpy,
+        },
+        {
+          provide: NavController,
+          useClass: NavMock,
```
Comment (OmkarJ13) [inline src/app/fyle/notifications/notifications.page.spec.ts:75]: Why not just use `useValue` and define the object?

File: src/app/fyle/notifications/notifications.page.spec.ts [hunk 3]
```diff
@@ -1,26 +1,210 @@
...
+    it('should only notify the delegatee', (done) => {
+      component.orgUserSettings$ = of(notificationDelegateeSettings1);
+
+      component.getDelegateeSubscription().subscribe((res) => {
+        expect(res).toEqual('Notify my delegate');
+        done();
+      });
+    });
+
+    it('should only the user', (done) => {
```
Comment (OmkarJ13) [inline src/app/fyle/notifications/notifications.page.spec.ts:124]: should only notify the user

File: src/app/fyle/notifications/notifications.page.spec.ts [hunk 4]
```diff
@@ -1,26 +1,210 @@
...
+      component.orgUserSettings$ = of(notificationDelegateeSettings2);
+
+      component.getDelegateeSubscription().subscribe((res) => {
+        expect(res).toEqual('Notify me only');
+        done();
+      });
+    });
+
+    it('should both user and delegatee', (done) => {
```
Comment (OmkarJ13) [inline src/app/fyle/notifications/notifications.page.spec.ts:133]: Rephrase here as well

File: src/app/fyle/notifications/notifications.page.spec.ts [hunk 5]
```diff
@@ -1,26 +1,210 @@
...
+    component.isAllEventsSubscribed();
+
+    expect(component.isAllSelected).toEqual({
+      emailEvents: false,
+      pushEvents: false,
+    });
+  });
+
+  it('updateNotificationEvents(): should update notification events', () => {
+    spyOn(component, 'removeAdminUnsbscribedEvents');
```
Comment (OmkarJ13) [inline src/app/fyle/notifications/notifications.page.spec.ts:161]: Let's fix the typo here

File: src/app/fyle/notifications/notifications.page.ts [hunk 6]
```diff
@@ -162,6 +162,7 @@ export class NotificationsPage implements OnInit {
   removeAdminUnsbscribedEvents(): void {
     this.orgSettings$.pipe(
       map((setting) => {
+        console.log(setting.admin_email_settings);
```
Comment (OmkarJ13) [inline src/app/fyle/notifications/notifications.page.ts:165]: Can you remove the console log before merging

File: src/app/fyle/notifications/notifications.page.spec.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/notifications/notifications.page.spec.ts:25]: This is a fix for `Nav Controller`, using a jasmine spy here does not work and causes some unknown observable error. It is likely that the controller is setting something up in the background causing this error. The mock class here is used to override the implementation.

File: src/app/fyle/notifications/notifications.page.spec.ts (hunk 2 above)
Comment (jayfyle) [reply src/app/fyle/notifications/notifications.page.spec.ts:75]: Same as above

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**